scripts/openmc-ace-to-hdf5

```diff
@@ -0,0 +1,205 @@
+#!/usr/bin/env python3
+
+import argparse
+import os
+import xml.etree.ElementTree as ET
+import warnings
+
+import openmc.data
+
+description = """
+This script can be used to create HDF5 nuclear data libraries used by
+OpenMC. There are four different ways you can specify ACE libraries that are to
+be converted:
+
+1. List each ACE library as a positional argument. This is very useful in
+   conjunction with the usual shell utilities (ls, find, etc.).
+2. Use the --xml option to specify a pre-v0.9 cross_sections.xml file.
+3. Use the --xsdir option to specify a MCNP xsdir file.
+4. Use the --xsdata option to specify a Serpent xsdata file.
+
+The script does not use any extra information from cross_sections.xml/ xsdir/
+xsdata files to determine whether the nuclide is metastable. Instead, the
+--metastable argument can be used to specify whether the ZAID naming convention
+follows the NNDC data convention (1000*Z + A + 300 + 100*m), or the MCNP data
+convention (essentially the same as NNDC, except that the first metastable state
+of Am242 is 95242 and the ground state is 95642).
+
+"""
+
+class CustomFormatter(argparse.ArgumentDefaultsHelpFormatter,
+                      argparse.RawDescriptionHelpFormatter):
+    pass
+
+parser = argparse.ArgumentParser(
+    description=description,
+    formatter_class=CustomFormatter
+)
+parser.add_argument('libraries', nargs='*',
+                    help='ACE libraries to convert to HDF5')
+parser.add_argument('-d', '--destination', default='.',
+                    help='Directory to create new library in')
+parser.add_argument('-m', '--metastable', choices=['mcnp', 'nndc'],
+                    default='nndc',
+                    help='How to interpret ZAIDs for metastable nuclides')
+parser.add_argument('--xml', help='Old-style cross_sections.xml that '
+                    'lists ACE libraries')
+parser.add_argument('--xsdir', help='MCNP xsdir file that lists '
+                    'ACE libraries')
+parser.add_argument('--xsdata', help='Serpent xsdata file that lists '
+                    'ACE libraries')
+parser.add_argument('--libver', choices=['earliest', 'latest'],
+                    default='earliest', help="Output HDF5 versioning. Use "
+                    "'earliest' for backwards compatibility or 'latest' for "
+                    "performance")
+args = parser.parse_args()
+
+if not os.path.isdir(args.destination):
+    os.mkdir(args.destination)
+
+# If the --xml argument was given, get the list of ACE libraries directory from
+# <ace_table> elements within the specified cross_sections.xml file
+ace_libraries = []
```

Comment:
Instead of being an array, it seems that you would like a `Set` here as all values in Ace_libraries need to be unique (per the usage). This would let you avoid certain `if` checks, and also be performant where needed due to Set operations.